tests/farmer_harvester/test_third_party_harvesters.py

```diff
@@ -0,0 +1,441 @@
+            cast(HarvesterAPI, harvester.server.api), request
+        )
+        assert result_msg is not None
+
+        # Inject overridden farmer reward address
+        response: RespondSignatures = dataclasses.replace(
+            RespondSignatures.from_bytes(result_msg.data), farmer_reward_address_override=farmer_reward_address
+        )
+
+        return make_msg(ProtocolMessageTypes.respond_signatures, response)
+
+    def validate_harvester_request_signatures(request: harvester_protocol.RequestSignatures) -> None:
+        nonlocal full_node_2
+        nonlocal farmer_reward_address
+        nonlocal validated_foliage_data
+        nonlocal validated_foliage_transaction
+        nonlocal validated_cc_vdf
+        nonlocal validated_rc_vdf
+        nonlocal validated_sub_slot_cc
+        nonlocal validated_sub_slot_rc
+        nonlocal finished_validating_data
+
+        assert request.message_data is not None
+        assert len(request.messages) > 0
+        assert len(request.messages) == len(request.message_data)
+
+        for hash, src in zip(request.messages, request.message_data):
+            assert hash
+            assert src
+
+            data: Optional[Streamable] = None
+            if src.kind == uint8(SigningDataKind.FOLIAGE_BLOCK_DATA):
+                data = FoliageBlockData.from_bytes(src.data)
+                assert (
+                    data.farmer_reward_puzzle_hash == farmer_reward_address
+                    or data.farmer_reward_puzzle_hash
+                    == bytes32(full_node_2.constants.GENESIS_PRE_FARM_FARMER_PUZZLE_HASH)
+                )
+                if data.farmer_reward_puzzle_hash == farmer_reward_address:
+                    validated_foliage_data = True
+            elif src.kind == uint8(SigningDataKind.FOLIAGE_TRANSACTION_BLOCK):
+                data = FoliageTransactionBlock.from_bytes(src.data)
+                validated_foliage_transaction = True
+            elif src.kind == uint8(SigningDataKind.CHALLENGE_CHAIN_VDF):
+                data = ClassgroupElement.from_bytes(src.data)
+                validated_cc_vdf = True
+            elif src.kind == uint8(SigningDataKind.REWARD_CHAIN_VDF):
+                data = ClassgroupElement.from_bytes(src.data)
+                validated_rc_vdf = True
+            elif src.kind == uint8(SigningDataKind.CHALLENGE_CHAIN_SUB_SLOT):
+                data = ChallengeChainSubSlot.from_bytes(src.data)
+                validated_sub_slot_cc = True
+            elif src.kind == uint8(SigningDataKind.REWARD_CHAIN_SUB_SLOT):
+                data = RewardChainSubSlot.from_bytes(src.data)
+                validated_sub_slot_rc = True
+            elif src.kind == uint8(SigningDataKind.PARTIAL):
+                # #NOTE: This data type is difficult to trigger, so it is
+                #        not tested for the time being.
+                # data = PostPartialPayload.from_bytes(src.data)
+                # validated_partial = True
```

Comment:
maybe you could add a word on why it's hard to test, and add a `no-cover` annotation